sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreCoordinator.scala

```diff
@@ -55,6 +56,39 @@ private case class GetLocation(storeId: StateStoreProviderId)
 private case class DeactivateInstances(runId: UUID)
   extends StateStoreCoordinatorMessage
 
+/**
+ * This message is used to report a state store has just finished uploading a snapshot,
+ * along with the timestamp in milliseconds and the snapshot version.
+ */
+private case class ReportSnapshotUploaded(
+    providerId: StateStoreProviderId,
+    version: Long,
+    timestamp: Long)
```

Comment:
Do we really need timestamp here ?

Reply:
I have this right now to reflect the actual upload timestamp more closely in case RPC calls get delayed, but since we also report the state store version loaded from checkpoints at the loading phase I figured this would be helpful in distinguishing between:

- reporting the snapshot version just uploaded vs 
- reporting the snapshot version just loaded but was uploaded at a previous unknown time

if this case is not necessary (since we have another way to detect cases from avail now) I can remove this if needed :+1:

Reply:
I guess if its only for logging purposes, its fine

Reply:
> I figured this would be helpful in distinguishing between:
> 
> * reporting the snapshot version just uploaded vs
> * reporting the snapshot version just loaded but was uploaded at a previous unknown time

How do we distinguish both? Do we put some marker value for the latter?

Reply:
This isn't distinguishable at the moment, we send the same RPC message for both types at the moment (treating the timestamp of loading the snapshot version as the upload timestamp).

During testing with some benchmark suites, a lot of state stores would be marked as lagging at the start because of the lack of time check (every store has default timestamp of 0, so it was easy to fulfill the time lag check). This is why both of those cases report a proper timestamp now.